postgresql/provider.go

```diff
@@ -130,17 +136,41 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	versionStr := d.Get("expected_version").(string)
 	version, _ := semver.Parse(versionStr)
 
-	config := Config{
-		Host:              d.Get("host").(string),
-		Port:              d.Get("port").(int),
-		Database:          d.Get("database").(string),
-		Username:          d.Get("username").(string),
-		Password:          d.Get("password").(string),
-		SSLMode:           sslMode,
-		ApplicationName:   tfAppName(),
-		ConnectTimeoutSec: d.Get("connect_timeout").(int),
-		MaxConns:          d.Get("max_connections").(int),
-		ExpectedVersion:   version,
+	var config Config
+	connString := d.Get("connection_string").(string)
```

Comment:
For the entire PR, can you make the following search/replace?  `s/connection_string/url/g`.  In the same breath, if you wanted to add `dsn` alongside `url`, that would be most welcome.